extension/goext/schemas.go

```diff
@@ -162,8 +162,8 @@ type ISchema interface {
 	// RegisterEventHandler registers an event handler for a named event with given priority
 	RegisterEventHandler(event string, handler func(context Context, resource Resource, environment IEnvironment) error, priority int)
 
-	// RegisterType registers a resource type, derived from BaseResource
-	RegisterType(resourceType interface{})
+	// RegisterType registers a resource type, derived from IResourceBase
+	RegisterType(resourceType IResourceBase)
 
 	// RegisterRawType registers a raw resource type, containing db annotations
 	RegisterRawType(rawResourceType interface{})
```

Comment:
perhaps this is not the perfect moment, but we should keep in mind, that similar issues may happen in `RegisterRawType`, `(Db)[Create|Update](Raw)` and return types of `(Lock)[Fetch|List](Raw)`

Reply:
Good idea - I'll create a separate PR for that.